docs/v3tov4.md

```diff
@@ -0,0 +1,340 @@
+      },
+      "metrics":[                             # list of metrics using the dimensional metric format
+        {
+          "name":"redis.metric1",
+          "type":"count",                     # gauge, count, summary, cumulative-count, rate or cumulative-rate
+          "value":93, 
+          "attributes":{}                     # set of key-value pairs that define the dimensions of the metric
+        }
+      ],
+      "inventory":{...},                      # Inventory remains the same
+      "events":[...]                          # Events remain the same
+    }
+  ]
+}
+```
+
+## GoSDK v4 API changes
+
+This section enumerates the main changes you have to keep in mind to upgrade from GoSDK v3.x to v4.x.
+
+### Entities
+
+In previous versions of the SDK, we had a distinction between the Local Entity and Remote entities. You could get the 
+`Local Entity` calling the method
+
+`func (i *Integration) LocalEntity() *Entity`
+
+or create a `Remote Entity` calling
+
+`func (i *Integration) Entity(name, namespace string, idAttributes ...IDAttribute) (e *Entity, err error)`
+
+In SDK v4, you can obtain the Host Entity or create Entities. The Host Entity is a the entity attached to the host 
+where the agent is running. It inherits the same entity ID as the agent which executes the integration. On the other
+hand, an Entity has its own entity ID. Both types of entities are decorated with the metadata from the host where the
+agent is running.
+
+To get the Host Entity you need to access the `HostEntity` of an integration. Example:
+
+```
+// create the integration
+i, err := integration.New(integrationName, integrationVersion)
+
+// add some inventory to the Host Entity
+i.HostEntity.AddInventoryItem("key", "value", "foo")
+```
+
+To create an Entity you need to call the following method inside  integration
+
+`func (i *Integration) NewEntity(name, type, displayName string) (e *Entity, err error)`
+
+Example: 
+
+```
+// create the integration
+i, err := integration.New(integrationName, integrationVersion)
+
+// create an entity
+entity, err := i.NewEntity(entityName, entityType, entityDisplayName)
+```
+
+These are the parameters accepted by `NewEntity`:
+
+* `entityName` must be an unique value per customer account because it uniquely identifies your Entity. It cannot be empty and 
+  the SDK cannot validate nor enforce that uniqueness, so it's up to the client to define a naming schema that produces 
+  unique names.
+
+  For example, using the following schema `<service_name>:<ip|hostname>:<port>`, we could create these names: 
+  `redis:192.168.100.200:12345` or `mysql:my_server_hostname:8086`
+
+  For the ip or the hostname, please avoid using the values `127.0.0.1` or `localhost` as they don't produce unique 
+  identifiers.
+
+* `entityType` describes the Entity's category. It cannot be empty. Some examples: "RedisInstance", "DockerContainer", 
```

Comment:
should we say what `entityType` is used for? (for NRQL queries and maybe for NR1 entity type grouping in the explorer(?))